python/paddle/fluid/tests/unittests/test_setitem_op.py

```diff
@@ -0,0 +1,204 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Test setitem op in static mode
+
+from __future__ import print_function
+
+import unittest
+import numpy as np
+
+import paddle
+
+
+class TestSetitemBase(unittest.TestCase):
+    def setUp(self):
+        paddle.enable_static()
+        self.set_value()
+        self.set_dtype()
+        self.shape = [2, 3, 4]
+        self.data = np.ones(self.shape).astype(self.dtype)
+        self.program = paddle.static.Program()
+
+    def set_value(self):
+        self.value = 6
+
+    def set_dtype(self):
+        self.dtype = "float32"
+
+    def _call_setitem(self, x):
+        x[0, 0] = self.value
+
+    def _get_answer(self):
+        self.data[0, 0] = self.value
+
+
+class TestSetitemApi(TestSetitemBase):
+    def test_api(self):
+        with paddle.static.program_guard(self.program):
+            x = paddle.ones(shape=self.shape, dtype=self.dtype)
+            self._call_setitem(x)
+
+        exe = paddle.static.Executor(paddle.CPUPlace())
+        out = exe.run(self.program, fetch_list=[x])
+
+        self._get_answer()
+        self.assertTrue(
+            (self.data == out).all(),
+            msg="\nExpected res = \n{}, \n\nbut received : \n{}".format(
+                self.data, out))
+
+
+# 1. Test different type of item: int, python slice
+class TestSetitemItemInt(TestSetitemApi):
+    def _call_setitem(self, x):
+        x[0] = self.value
+
+    def _get_answer(self):
+        self.data[0] = self.value
+
+
+class TestSetitemItemSlice(TestSetitemApi):
+    def _call_setitem(self, x):
+        x[0:2] = self.value
+
+    def _get_answer(self):
+        self.data[0:2] = self.value
+
+
+class TestSetitemItemSlice2(TestSetitemApi):
+    def _call_setitem(self, x):
+        x[0:-1] = self.value
+
+    def _get_answer(self):
+        self.data[0:-1] = self.value
+
+
+class TestSetitemItemSlice3(TestSetitemApi):
+    def _call_setitem(self, x):
+        x[0:-1, 0:2] = self.value
+
+    def _get_answer(self):
+        self.data[0:-1, 0:2] = self.value
+
+
+class TestSetitemItemSlice4(TestSetitemApi):
+    def _call_setitem(self, x):
+        x[0:, 1:2, :] = self.value
+
+    def _get_answer(self):
+        self.data[0:, 1:2, :] = self.value
+
+
+# 2. Test different type of value: int, float, numpy.ndarray, Tensor
+
+
+def create_test_value_numpy(parent):
+    class TestValueInt(parent):
+        def set_value(self):
+            self.value = np.array([1])
+
+    cls_name = "{0}_{1}".format(parent.__name__, "ValueInt")
+    TestValueInt.__name__ = cls_name
+    globals()[cls_name] = TestValueInt
+
+
+create_test_value_numpy(TestSetitemItemInt)
+create_test_value_numpy(TestSetitemItemSlice)
+create_test_value_numpy(TestSetitemItemSlice2)
+create_test_value_numpy(TestSetitemItemSlice3)
+create_test_value_numpy(TestSetitemItemSlice4)
+
+
+def create_test_value_tensor(parent):
```

Comment:
It seems `create_test_value_tensor` only test int Tensor why you commented test different types above?

Reply:
Thanks, added unittests for different dtypes.